plugin/pkg/auth/authenticator/request/x509/x509.go

```diff
@@ -52,28 +52,34 @@ func New(opts x509.VerifyOptions, user UserConversion) *Authenticator {
 
 // AuthenticateRequest authenticates the request using presented client certificates
 func (a *Authenticator) AuthenticateRequest(req *http.Request) (user.Info, bool, error) {
-	if req.TLS == nil {
+	if req.TLS == nil || len(req.TLS.PeerCertificates) == 0 {
 		return nil, false, nil
 	}
 
+	// Use intermediates, if provided
```

Comment:
see https://golang.org/src/crypto/tls/handshake_server.go, processCertsFromClient for how the stdlib verifies the cert chain

only the leaf cert is verified, the remaining certs are client-supplied intermediates